### PR TITLE
fix: Move persistence layer's delta computation into worker

### LINKED ISSUE #5561: Move persistence layer's delta computation into worker
Our profiling shows that, for a large bot, the file persistence layer's delta computation (calculating the delta to be saved to disk\storage) is taking a fair amount time and could cause the UI thread to be blocked. 

image

The 4th in the above picture is showing file persistence layer is taking about 100ms. 

The proposal is moving this part into worker, so that UI thread won't be blocked. 

Note: the reason we are calculating the state change between commits is we don't have a native support in recoil **yet**.  If we get a native support, we could save the whole delta computation process.

### PR DESCRIPTION
**Pull request by lei9444**
## Description
The persistence layer will check files' change after the state changed. When we use a big bot such as the [calendarSkill](https://github.com/microsoft/botframework-components/tree/calendarSkill_v2/skills/declarative/calendarSkill), the check function will block the main thread about 90ms every notify. 

**fix** 
Move the complex computation to worker

## Task Item
closes #5561 

## Screenshots
**before:**
image

**after:**
image

### CODE REVIEW

**Comment by boydc2014** [comment]
I can see the differences, but do we know why after moving to worker, we still take about 10ms?

**Comment by lei9444** [comment]
> I can see the perf gain here, just one more question, do we know why after moving to worker, we still take about 10ms?

The worker.postMessage will take some time here

**Comment by srinaath** [comment]
This is awesome work @lei9444 ! Moving it a worker is so much better.

**Review by srinaath — APPROVED** [review]

File: Composer/packages/client/src/recoilModel/persistence/types.ts [hunk 1]
```diff
@@ -30,3 +41,19 @@ export interface IFileChange {
...
+  | DialogInfo
+  | DialogSchemaFile
+  | LuFile
+  | QnAFile
+  | LgFile
+  | SkillManifestFile
+  | RecognizerFile
+  | FormDialogSchema;
+
+export type Difference = {
```
Comment (srinaath) [inline Composer/packages/client/src/recoilModel/persistence/types.ts:55]: Myabe name could be FileDifference?

File: Composer/packages/client/src/recoilModel/parsers/calculator.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+// Wrapper class
+class Calculator extends BaseWorker<CalculatorType> {
+  difference(target: FileAsset[], origin: FileAsset[]) {
+    const payload = { target, origin };
+    return this.sendMsg<FilesDifferencePayload>('difference', payload);
+  }
+}
+
+export default new Calculator(new Worker());
```
Comment (srinaath) [inline Composer/packages/client/src/recoilModel/parsers/calculator.ts:18]: FileDiffCalculator?

**Review by boydc2014 — APPROVED** [review]

**Pull request merged**